.github/workflows/deploy.yml

```diff
@@ -1,36 +1,43 @@
-# This is a basic workflow to help you get started with Actions
-
 name: CI
 
-# Controls when the workflow will run
 on:
   # Triggers the workflow on push or pull request events but only for the master branch
   push:
     branches: [master]
 
+  pull_request:
+    branches: [master]
+
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
   # This workflow contains a single job called "build"
   build-and-deploy:
-    # The type of runner that the job will run on
+    name: Build and Deploy
     runs-on: ubuntu-latest
 
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v3
 
-      - name: Install and Build 🔧 # This example project is built using npm and outputs the result to the 'build' folder. Replace with the commands required to build your project, or remove this step entirely if your site is pre-built.
-        run: |
-          yarn install
-          yarn build
-          yarn deploy
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "18"
+          cache: "yarn"
+
+      - name: Install dependencies
+        run: yarn install --immutable
+
+      - name: Build
+        run: yarn build
+
+      - name: Deploy
+        run: yarn deploy
 
-        # --legacy-peer-deps is used due to a dependency between "@vue/cli-plugin-eslint" and eslint
-      - name: Deploy 🚀
+      - name: Deploy to github pages 🚀
+        if: github.ref == 'refs/heads/master'
```

Comment:
What's the problem solved by the `if: github.ref == 'refs/heads/master'`

Reply:
The one you mentioned in the other comment. The deployment to github pages is done only when running from the master branch, not from a PR.

Reply:
In fact you can see that it has been skipped for this PR

Reply:
> you can see that it has been skipped for this PR

Not 100% sure, but I think GitHub run actions according to the main branch. The only way I know to test actions is to push it to the main branch of your forked repository, and open a PR on it.


I still don't get the problem you're solving. For me, you do two things:
- trigger deploy action when opening a PR on `master` by adding
    ```yaml
    pull_request:
      branches: [master]
    ```
- block the deployment by adding 
    ```yaml
    if: github.ref == 'refs/heads/master'
    ```

Notice that no `pull_request` could get this `github.ref`, because for PR ref is `refs/pull/<pr_number>/merge`

> The ref given is fully-formed, meaning that for branches the format is `refs/heads/<branch_name>`, for pull requests it is `refs/pull/<pr_number>/merge`, and for tags it is `refs/tags/<tag_name>`. For example, refs/heads/feature-branch-1.

From: https://docs.github.com/en/actions/learn-github-actions/contexts#github-context


----

Why not removing both 
```yaml
pull_request:
  branches: [master]
```

and

```yaml
if: github.ref == 'refs/heads/master'
```

Reply:
This is useful to test for changes before merging with the same workflow (except deploying) that we use for a real deployment. I have no hard view about it, so if you want it I can remove it. Also about the test, if you want we can try in my repo. Let me know!